### ISSUE TITLE
parallel treatment

### ISSUE THREAD

**Issue by eroux** [buda-base/volume-manifest-builder#3]
We should be able to define several jobs running in parallel, the current script is too slow

**Comment by eroux** [comment]
https://pypi.org/project/aioboto3/ looks reasonable

**Comment by jimk-bdrc** [comment]
It would be nice to parallelize the internal functions. Encapsulate the data you extract, and pass that to a separate task to dispatch the manifest and the errors. That way we could control the output better.

**Comment by jimk-bdrc** [comment]
As well, I'm automating this into a service which is installed on an EC2 launch template, which the sync service kicks off. The service reads its input from an S3 object (manifest.bdrc.org/processing/todo), processes the works in the list, saves its diagnostics and logs to S3, and then shuts down the instance. (We set the launch template to terminate on shutdown).
_update_ learning about CloudFormation. There could be a good way to sync these together behind an AWS Queue.

**Comment by eroux** [comment]
The s3 workflow looks very excellent! I'm not quite sure what you mean in your first comment though? Can you elaborate a little bit?

**Comment by jimk-bdrc** [comment]
Sure. Right now, v-m-t bring down a blob of image blob, and then use the same method to 
+ extract the sizes
+ detect errors
+ write the errors. 

It would be better to pass the blob to a set of dedicated routines each of which do one thing. In that way , we could run those dedicated routines in parallel. (I'm strongly considering this in Java in Audit Tool.)

**Comment by jimk-bdrc** [comment]
HI @eroux, speed is not a factor in this script right now, so I'm going to close this. Even if I were to parallelize it, I would use Python Threading and queues - they are much easier to program.
